types/assertions.d.ts

```diff
@@ -137,6 +137,12 @@ export interface DeepEqualAssertion {
 	 * `expected`, returning a boolean indicating whether the assertion passed.
 	 */
 	<Actual, Expected extends Actual>(actual: Actual, expected: Expected, message?: string): actual is Expected;
+	
+	/**
+	 * Assert that `actual` is [deeply equal](https://github.com/concordancejs/concordance#comparison-details) to
+	 * `expected`, returning a boolean indicating whether the assertion passed.
+	 */
+	<Actual extends Expected, Expected>(actual: Actual, expected: Expected, message?: string): expected is Actual;
```

Comment:
Actually this might not want to be a type guards. It might just simply want to return a boolean.

Reply:
I'll figure out this when writing the tests.